sympy/printing/str.py

```diff
@@ -74,13 +74,16 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+    def _print_Not(self, expr):
+        return '~(%s)' %(expr.args[0])
+
     def _print_And(self, expr):
-        return '%s(%s)' % (expr.func, ', '.join(sorted(self._print(a) for a in
-            expr.args)))
+        return '%s' %(' & '.join([self.parenthesize(arg, 38)
```

Comment:
Shall I put in a comment like this : 
```python
# The number 38 has been used (instead of PRECEDENCE["And"]) as 
# Bitwise and operator (&) is used to represent And objects in SymPy
# and not Boolean and (and).
```